### PR TITLE
Feature/issue 185/location based viewing

### PR DESCRIPTION
**Pull request by QahtanAlJammali**
# [Issue #185  ](https://github.com/gcivil-nyu-org/wed-fall24-team5/issues/185)

## Description
Added automatic location detection and default searching of donations that are 5 miles away. Added search based on any custom location with different radius to search around. Added the distance of each organization from the location. 
Added geopy to the requirements.txt since it is used for the location related services. Please see image below for the additions in the dashboard.
 
Replace this with a summary of your changes including any high-level business or decision-making context.
Screenshot 2024-11-03 at 12 39 47 PM

## Change Type (delete non-relevant options)

- 💡 New feature (non-breaking change which adds functionality)
- 🧪 Test (for adding tests)

## Dependencies

- No dependancies

### CODE REVIEW

File: src/recipient_dashboard/tests.py [hunk 1]
```diff
@@ -480,3 +466,82 @@ def test_search_no_results(self):
...
+    response = self.client.get(url)
+
+    # Check that no items are returned
+    self.assertEqual(len(response.context["donations"]), 0)
+
+
+# In a new test class for utils.py
+
+
+class UtilsTests(TestCase):
```
Comment (arsalananwar11) [inline src/recipient_dashboard/tests.py:517]: Any specific reason for adding this class?

**Review by arsalananwar11 — COMMENTED** [review]
Looks really great Qahtan! Would love to see this production. Thanks for implementing this. Also, for some reason, the travis build fails, I tried rerunning but the coveralls triggers some issues. Can you please check?

File: src/recipient_dashboard/tests.py (hunk 1 above)
Comment (QahtanAlJammali) [reply src/recipient_dashboard/tests.py]: Removed. I had it as a place holder to test urls.py

**Comment by QahtanAlJammali** [comment]
> Looks really great Qahtan! Would love to see this production. Thanks for implementing this. Also, for some reason, the travis build fails, I tried rerunning but the coveralls triggers some issues. Can you please check?

Thanks sir, it was building ok on travis but the coverage went down, so I was creating tests, but the tests were having some issues, since testing location based items can be quite tricky. I'll keep working on increasing test coverage so that I can merge.

**Comment by QahtanAlJammali** [comment]
@arsalananwar11 If we merge this to see the functionality in production, I can create another issue separately specifically for test coverage.

**Comment by QahtanAlJammali** [comment]
Increased the coverage, now its only down -0.7% Would this please be merged and I'll work on increasing coverage in other areas

File: .gitignore
Comment (QahtanAlJammali) [inline .gitignore:1]: @arsalananwar11 removed personal file from gitignore and directory.

**Review by arsalananwar11 — APPROVED** [review]
Looks great, thanks for the changes Qahtan

**Pull request merged**